api/segmentation/class-segmentation-client-data.php

```diff
@@ -50,6 +54,38 @@ public function report_client_data( $request ) {
 			$client_data_update['email_subscriptions'][] = $email_subscription;
 		}
 
+		// Fetch Mailchimp data.
+		$mailchimp_campaign_id   = $this->get_request_param( 'mc_cid', $request );
+		$mailchimp_subscriber_id = $this->get_request_param( 'mc_eid', $request );
+		if ( $mailchimp_campaign_id && $mailchimp_subscriber_id ) {
+			$mailchimp_api_key_option_name = 'newspack_newsletters_mailchimp_api_key';
+			global $wpdb;
+			$mailchimp_api_key = $wpdb->get_row( // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
```

Comment:
@adekbadek Just wanted to ask, are we accessing the options table directly for performance reasons, as opposed to just using `get_option`?

Reply:
This is because in the API part of this plugin, WordPress is not loaded. Just the bare minimum is, and some functions are mocked – see `api/setup.php`.